### ISSUE TITLE
Make it easier to discover the targets that you can reference when reading a MyST site

### ISSUE THREAD

**Issue by choldgraf** [executablebooks/mystmd#454]
### Which area is this feature request for?

Website User Interface

### Describe the feature you'd like to request

When **referencing external projects**, you need to know the "target ID" of the thing to reference in order to incorporate it in your MyST reference (e.g., `[](#myid)`). However you're generally not interacting with their source code, but are instead reading a rendered HTML version of the page.

It would be helpful if there were a way to quickly look up the IDs that already exist on a published page *from the browser* so that you could quickly find this information.

It would be even more interesting if these IDs were generated in a unique fashion regardless of whether the author added them manually. This would mean you could pick an arbitrary paragraph from any published MyST site and reference it from another one.

**Comment by rowanc1** [comment]
Really like this idea!

Right now the IDs that are stable are in the users control, e.g. by labeling an equation or a header. I have been thinking about this a bunch but I have no idea how to generate stable IDs across builds (event relatively stable + unique seems hard). Any thoughts on that?

For the UI, I am hesitant to put in too many visual anchors, and we have been moving away from putting things in the margins. Should this be a context menu? Or any other ideas of how this should look?

**Comment by rowanc1** [comment]
There is a small little command-line tools that allows you to search, but that is still pretty technical.

```bash
npm install -g intersphinx
intersphinx list https://mystmd.org/guide --limit 5 --includes frontmatter
```

![image](https://github.com/executablebooks/mystmd/assets/913249/338e90c7-1ddd-4466-86c0-b800a07699d4)

**Comment by choldgraf** [comment]
Indeed - from an ideal workflow perspective, I'd want to do this via the website directly. Similar to how you can hover over an H1, see the little "link" `#` pop up, and immediately click it to have a link. I don't know how you'd let this happen for arbitrary paragraphs without _really_ cluttering up the UI though. Maybe if it were restricted to the most commonly-referenced things that you already get unique `id=` values for? 🤔

**Comment by rowanc1** [comment]
Having something similar to github might be ok? It is actually hidden most of the time until you click on the line. The UI has to leave space for the left hand side.

![image](https://github.com/executablebooks/mystmd/assets/913249/1efd3283-300d-40ca-bf36-0e89759fbf26)

Notion is even more subtle:

![image](https://github.com/executablebooks/mystmd/assets/913249/9c82855d-5b6f-446e-a1ef-3033b4e0819e)

**Comment by choldgraf** [comment]
Yep that is the kinda thing that I had in mind as well, it'd need to be very non-intrusive but some way to "expand" what is available to a block of content might be really neat

**Comment by JimMadge** [comment]
Adding my +1 to this feature. When referencing an external project I've used a combination of guessing, and looking through the git repository to find labels, which is a bit slow and inconvenient.

**Comment by agoose77** [comment]
@JimMadge I think this is a vital feature!
